apollo-compiler/src/test/graphql/com/example/unique_type_name/HeroDetailQueryExpected.java

```diff
@@ -61,9 +61,9 @@ public String queryDocument() {
   }
 
   public interface Data extends Operation.Data {
-    @Nullable Hero hero();
+    @Nullable HeroDetailQuery$ heroDetailQuery();
```

Comment:
Not sure why we initially decided to go with $, but it's really easy to change in compiler. But the idea that you in most cases you won't have such issue with more than 2-3 nested classes with the same name. But I don't have any strong opinion for this as Hero4 and Hero$$$$ from my perspective is the same evil, and for sure I will try to avoid such queries with the same nested names (will use aliases)